framework/SwiftPlot/Series.swift

```diff
@@ -1,28 +1,38 @@
 // struct defining a data series
-public struct Series {
+public struct Series<T,U> {
     public var barGraphSeriesOptions = BarGraphSeriesOptions()
     public var scatterPlotSeriesOptions = ScatterPlotSeriesOptions()
-    public var points = [Point]()
-    public var scaledPoints = [Point]()
+    public var pairs = [Pair<T,U>]()
+    public var scaledPairs = [Pair<T,U>]()
     public var maxY: Float = 0
     public var minY: Float = 0
     public var label = "Plot"
     public var color : Color = .blue
     public var startColor: Color? = nil
     public var endColor: Color? = nil
     public init() {}
-    public init(points p: [Point], label l: String, color c: Color = Color.lightBlue, hatchPattern: BarGraphSeriesOptions.Hatching = .none, scatterPattern: ScatterPlotSeriesOptions.ScatterPattern = .circle){
-        points = p
+
+    public init(pairs : [Pair<T,U>],
+                label l: String,
+                startColor : Color = .lightBlue,
+                endColor : Color = Color.lightBlue,
+                hatchPattern: BarGraphSeriesOptions.Hatching = .none,
+                scatterPattern: ScatterPlotSeriesOptions.ScatterPattern = .circle){
+        self.pairs = pairs
```

Comment:
Thinking about this further, I don't think `pairs` (or `values` or whatever) should be exposed directly within Series. Instead, Series should act as a collection that wraps this. For example, `count` should be defined directly on Series so that it returns `pairs.count`, which would simplify code in a few places.

Likewise, you should define a custom subscript operator on Series that indexes into `pairs`. Finally, per my earlier comments on getMaxX, maybe you just have a maxX() function on Series itself and save having to call into the pairs property externally.

Reply:
Yes! This is a great idea. It would make it more convenient to use the Series type.

Reply:
@BradLarson if we do not expose `values` from Series will we want to have series as the variable to be passed into maxX and maxY functions? 
I think letting it be exposed along with adding the subscript operator might be a good idea? We might need to make a copy of just the `values` and not the entire series variable...

Reply:
You could try both, and see what makes sense. I've generally wrapped values like this so that this implementation detail isn't exposed, but see what works best for your architecture.

About HistogramSeries, is there a different way to architect that so that you could use a more generic Series instead of having a special type for that? Could a histogram process a Series into a more specific internal representation? How do other frameworks handle this, do people provide a more standard set of datapoints and have the histogram plot process them, or do they provide histogram-specific data?

Reply:
The kind of data just a little different. Instead of having X and Y values, we need the user to give just an array of data and the number of bins it needs to be divided into. I wrote a different Series type because we don't need the X vales here. I'll try if I can use the normal Series type here.